### PR TITLE
RL-SM: Add Functionality for onBuy, onSell Buttons, updating totalWealth using 2 PUT endpoints

### PR DESCRIPTION
**Pull request by Lin2**
# Overview
We added two PUT endpoints for BUY and SELL, and created onBuy and onSell basic functionality to update totalWealth of a user on the Playpage.

# Issues Addressed
 We addressed some of the points in #15 . This issue isn't completed yet however, but we wanted to get a PR in for CI/CD.

We addressed these issues:

 * PlayPage.js updates TotalWealth (located in backend UserCommons.java) in onBuy and onSell based on Market Cow Price.
 * Get FarmStats.js to display Total Wealth.
 * UserCommonsController.java has 2 PUT endpoints, for buy and sell
 * Buy and Sell Endpoints have USER authorization
 * Mutation testing passes
 
These are still left for future work

* PlayPage.js calls backend to update Average Cow Health in onBuy and onSell using given equations. For selling, Cows are sold for user averageCowHealth * cowPrice
*  Make FarmStats.js to display average cow health
 * 100% test coverage for frontend

# Details

When a user presses the "+"(onBuy) or "-"(onSell) button on the play page on the website, we call the 2 PUT endpoints /buy and /sell. These buy and sell 1 cow respectively each time they're called. Currently, it sells at cowPrice (value of which comes from when an admin creates a new commons), and buys at cowPrice (although this needs to be changed in a future edit to factor in average cowHealth).

User totalWealth comes from startingBalance, which is set by an admin when creating a new commons.

We created 4 tests for the buy and sell endpoint (2 of which are for checking when the usercommons doesn't exist).

We deleted a line of code from a backend test created by w22-6pm-4 (see commit message regarding it).

# Testing Plan

1. Create a commons
2. Join the commons 
3. The Total Wealth under Your Farm Stats should be the Starting Wealth entered when you created the commons
4. When "+" is clicked under Manage Cows, Total Wealth decreases by the Cow Price entered when you created the commons
5. When "-" is clicked under Manage Cows, Total Wealth decreases by the Cow Price entered when you created the commons

### CODE REVIEW

**Review by pconrad — CHANGES_REQUESTED** [review]
Where are the controller tests for the new methods?

**Comment by Lin2** [comment]
> Where are the controller tests for the new methods?

I realized we were missing this right after we made the request -- working on it now!

**Comment by Lin2** [comment]
The PR has been completed with controller tests for UserCommonsController.

**Review by eambriz27 — DISMISSED** [review]
Looks like you have some merge conflicts to resolve, but everything else LGTM!

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java [hunk 1]
```diff
@@ -96,36 +99,323 @@ public void test_getUserCommonsById_nonexists_admin() throws Exception {
...
+
+
+    String requestBody = mapper.writeValueAsString(userCommonsToSend);
+    String expectedReturn = mapper.writeValueAsString(correctuserCommons);
+
+    when(userCommonsRepository.findByCommonsIdAndUserId(eq(1L), eq(1L))).thenReturn(Optional.of(origUserCommons));
+    when(commonsRepository.findById(eq(1L))).thenReturn(Optional.of(randomCommons));
+
+    // act
+    MvcResult response = mockMvc.perform(put("/api/usercommons/buy?id=1") //oof, accidentally made PUT endpoints id variable instead of commonsId. TODO: Fix later.
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java]: Fix now maybe?

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java [hunk 2]
```diff
@@ -96,36 +99,323 @@ public void test_getUserCommonsById_nonexists_admin() throws Exception {
...
+
+
+
+
+
+
+
+
+
+
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java]: As long as we are fixing things, maybe this white space too...

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java [hunk 3]
```diff
@@ -96,36 +99,323 @@ public void test_getUserCommonsById_nonexists_admin() throws Exception {
...
+    String requestBody = mapper.writeValueAsString(userCommonsToSend);
+    String expectedReturn = mapper.writeValueAsString(correctuserCommons);
+
+    when(userCommonsRepository.findByCommonsIdAndUserId(eq(1L), eq(1L))).thenReturn(Optional.of(origUserCommons));
+    when(commonsRepository.findById(eq(1L))).thenReturn(Optional.of(randomCommons));
+
+    // act
+    MvcResult response = mockMvc.perform(put("/api/usercommons/buy?id=234") //oof, accidentally made PUT endpoints id variable instead of commonsId. TODO: Fix later.
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java]: oof indeed :-)

**Review by pconrad — CHANGES_REQUESTED** [review]
Couple of "TODOs left in the code"...

**Review by eambriz27 — DISMISSED** [review]
LGTM

**Comment by Lin2** [comment]
Bad PR

**Pull request closed without merging**